### PR TITLE
Supermatter Shards Can No Longer Be Invisible Sprayed

### LINKED ISSUE #13841: Invisbility spray can be used on the supermatter shard and singularity
A bit much perhaps
image

### PR DESCRIPTION
**Pull request by Shadowmech88**
Fixes #13841

:cl:
 * tweak: The supermatter shard will now fuck you up if you apply invisible spray to it.

### CODE REVIEW

**Comment by F-OS** [comment]
Rip fun.

**Comment by ScrambledSteak** [comment]
honk is dead

no honk

**Comment by Probe1** [comment]
I like how I open the game and hear people bitch about it.  I look at what it is  and it's clearly absurd.  Spraying it at all should irradiate the fuck out of you and possibly kill you.

But then I open github and hear about how its funny and no this is classic comedy.

You guys wonder why I sometimes don't treat your emojis like divine mandate.  This is why.

**Comment by N3X15** [comment]
Also this was considered during the invisibility spray PR and was rejected because it would, in fact, actually be funny.

**Comment by Probe1** [comment]
It turned out players didn't actually find it funny to walk into an invisible insta kill shard.

**Comment by BobdaBiscuit** [comment]
People walk into the ~~~in~~~visible insta kill shard anyway.
Hiding one in the same tile as a closed door is always comedy gold.

**Comment by mtpowell** [comment]
>adds shard but not singularity

rev up them commits

**Comment by Probe1** [comment]
I ignored that @mtpowell because it actually is kinda impressive and funny to make the singlo vanish.

You will probably notice the station being destroyed and turn the other direction.

**Comment by mtpowell** [comment]
@Probe1 
It's not too impressive, you can paint it invisible when it's stage one and doesn't pull/radiate. I'd argue it's even easier than the shard.

**Comment by Probe1** [comment]
Yeah you're right. I acknowledged that and wrote it off but really it's the exact same bullshit in the end.

**Comment by clusterfack** [comment]
Why does this use preattack? This all would have been avoided if you had used afterattack instead.

**Comment by Probe1** [comment]
I assumed so he could avoid having clowns shard themselves.

**Comment by Shadowmech88** [comment]
Actually it was to prevent "You attack the X with the can of invisible spray!" messages.

**Comment by Kitfox88** [comment]
>I assumed so he could avoid having clowns shard themselves.

sacrifices must be made for the sake of honks

File: code/game/objects/items/weapons/cosmetics.dm [hunk 1]
```diff
@@ -215,6 +218,10 @@
 	if(target.invisibility || target.alpha <= 1)
 		to_chat(user, "\The [target] is already invisible!")
 		return
+	for(var/i in prohibited_objects)
```
Comment (PJB3005) [inline code/game/objects/items/weapons/cosmetics.dm]: Use `is_type_in_list` instead.

File: code/game/objects/items/weapons/cosmetics.dm [hunk 2]
```diff
@@ -203,6 +203,9 @@
 	var/permanent = 0
 	var/invisible_time = 5 MINUTES
 	var/sprays_left = 1
+	var/list/prohibited_objects = list( //For fun removal
```
Comment (PJB3005) [inline code/game/objects/items/weapons/cosmetics.dm]: Make this `/static`

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by SonixApache** [comment]
Making the spray activate the shard (but still make it invisible) might be a better move.

**Comment by ghost** [comment]
After personally dealing with one of these things, I kinda disagree with making it impossible as it is possible to deal with.

Should ash the fucking clown that does it, but don't make it impossible

**Comment by StreptoMutans** [comment]
You didn't listen

**Comment by gbasood** [comment]
I think this is an important change but I will also miss it
Possibly @Skullyton is on the right track instead

**Comment by Shadowmech88** [comment]
* The supermatter shard is no longer exempted from being invisible sprayed, however doing so will now cause you to touch the can to the shard, with results the same as touching the shard with any item.

**Comment by Sprok0** [comment]
Touching the shard with an item just deletes the item and irradiates you, sprays are already one-use and you'll almost always have a radsuit if attempting this in the first place.

**Review by PJB3005 — APPROVED** [review]

**Comment by Killette2** [comment]
Greytide implant + invisible spray
"Clown, spray that shard for me"

**Pull request merged**